scripts/abi_artifact_generator/end_to_end.md

```diff
@@ -0,0 +1,308 @@
+# The end to end example of what happens within the testing suite
```

Comment:
Give a brief overview stating that `randomized_state` is used to initialize the CPU state in a known way and parts of it will be copied into `printable_location` by the code generated by the compiler and then another tool downstream will check if `printable_location` contains what we expect.

Reply:
Done.